strategy/kaitiaki.js

```diff
@@ -0,0 +1,57 @@
+const isEqual = require('lodash.isequal')
+
+const IDENTITY = {
+  add: [],
+  remove: []
+}
+
+function KaitiakiRule () {
+  // Entry = { feed: String, seq: Integer }
+  // transformations are of form:
+  // - { add: [], remove: []} (identity)
+  // - { add: [Entry], remove: [Entry] }
+  // - { add: [Entry], remove: [] }
+  // - { add: [], remove: [Entry] }
```

Comment:
I think this should definitely be the "API" for how users enter the transformations in e.g. `ssb-profile`.
What I'm not sure of is whether this format will work when we try to write concat.

one challenge with continuing to use arrays is that for the `(a*b)*c === a*(b*c)` rule (associativity) to work, you will have to constantly be sorting the Arrays to ensure consistency. Not the end of the world, but could be easier.

_VERY_ rough sketch of another idea for how to store transformations:

```js
{
  @ben: {
    200: 1,  // ben added at seq 200
    550: -1,  // removed at seq 550 (by cherese)
    500: -1,  // removed at seq 500 (by mix who was out of the loop)
    700: 1
  },
  @cherese: {
    1400: 1
  },
}
```

the reified state would come out like:
```js
{
  @ben: [{ start: 200, end: 500 }, { start: 700, end: null }],
  @cherese: [{ start: 1400, end: null }]
}
```

note that in my example there's a redundent removal and the system doesn't care when we reify. it's important that that data isn't lost in the transformation though because it help guarentee associativity, and I _think_ in this case "commutativity" i.e. `a*b === b*a` which is _real nice_ in this context because with those two things I think it's basically impossible for there to be any conflicts...

Reply:
I like this way much better than the other, do you think we could somehow reuse the simple-set strategy in this? seems like the object looks similar to a simple set

Reply:
```
@feed: {
  Integer: Integer
}
```

Reply:
it's different to the simple-set in that that one doesn't have this depth.
that one has 

```
{
  @ben: 1
}
```
which only tells you if something is in / out of the set of authors. We need to know _when_ they were in / out

Reply:
Oh right!